subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -23,6 +23,7 @@
 #include "lll_adv.h"
 #include "ull_adv_types.h"
 #include "ull_adv_internal.h"
+#include "lll_adv_internal.h"
```

Comment:
`lll_*_internal.h` should not be used in ULL files. I am working towards correcting it in https://github.com/zephyrproject-rtos/zephyr/pull/32613

See if you can split out vendor specifics in `lll_adv_internal.h`

Reply:
The `#iclude lll_adv_internal.h` is required due to a single function; `lll_adv_sync_data_enqueue`. I can try to do a similar change as you did in https://github.com/zephyrproject-rtos/zephyr/pull/32613 but I do not want to change partially use of of `lll_xxx_internal.h` only partially. 

If I start to extract code that in your PR landed in `lll_adv_pdu.h` I'll be forced to change other files also. 
And it is not part of this commit goal.

In my humble opinion it is better to correct the use of header files in single PR (for all source files in controller).
If you strongly advise to do remove the `#iclude lll_adv_internal.h` from the `ull_df.c` I'll try to deliver it.

Reply:
> See if you can split out vendor specifics in lll_adv_internal.h

No, this was just a suggestion. Yes, can wait for a separate PR to address this, based on the outcome of #32613.